log/log.go

```diff
@@ -39,10 +41,63 @@ func (f levelFlag) Set(level string) error {
 	return nil
 }
 
+func setSyslogFormatter(appname, local string) error {
+	if appname != "" && local != "" {
+		return nil
+	}
+
+	fmter, err := newSyslogger(appname, local, origLogger.Formatter)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error creating syslog formatter: %v\n", err)
+		origLogger.Errorf("can't connect logger to syslogs: %v", err)
+		return err
+	}
+	origLogger.Formatter = fmter
+	return nil
+}
+
+func setJSONFormatter() {
+	origLogger.Formatter = &logrus.JSONFormatter{}
+}
+
+type logFormatFlag struct{}
+
+// String implements flag.Value.
+func (f logFormatFlag) String() string {
+	return origLogger.Level.String()
```

Comment:
This now should return something different than the log level? Guess this was a copy-and-paste from the `levelFlag.String()` implementation?

See also the wrong default values in the flag help output now:

```
   -log.format "info"
      If set use a syslog logger or JSON logging. Example: 
      logger:syslog?appname=bob&local=7 or logger:stdout?json=true
  
   -log.level "info"
      Only log messages with the given severity or above. Valid levels: 
      [debug, info, warn, error, fatal].
```